internal/dag/dag.go

```diff
@@ -293,6 +293,16 @@ func (s *Secret) Data() map[string][]byte {
 	return s.Object.Data
 }
 
+// Cert returns the secret's tls certificate
+func (s *Secret) Cert() []byte {
+	return s.Object.Data[v1.TLSCertKey]
```

Comment:
Could there be a case where the cert is not in the secret?

Reply:
It's plausible.  It is assumed to be valued when setting up the downstream context; but on review, that isn't guaranteed. 
 - `envoy.DownstreamTLSContext(data[v1.TLSCertKey], data[v1.TLSPrivateKeyKey], vh.MinProtoVersion, alpnProtos...)`
 - https://github.com/heptio/contour/blob/master/internal/contour/listener.go#L265

It feels like the the structure should be verified while creating a `SecureVirtualHost` vertex.

Reply:
These got marked as outdated when I rebased. They were brought in with #1047.

Reply:
ok no biggie, there's another issue where we get the cert and not the key and it causes issues, but let's not try and fix that in this (just been on my mind).

Reply:
Its okay! I'm glad you mentioned it.  I remember looking at it while I drafted the PR, when trying to decide on introducing the helper functions on *dag.Secret.  Is there a already an issue for that?  I can clean them all up at one time in another PR :).

Reply:
I bet its this one, #1051  :)